Update the documentation on installing with Typst

choldgraf opened the pull request:
This adds clearer instructions for how to install and use Typst, see [this discord thread for some context](https://discord.com/channels/1083088970059096114/1355534877553983529/1356388423807537362)

choldgraf commented on docs/creating-pdf-documents.md:
```diff
@@ -93,9 +94,31 @@ Ensure that you download a full distribution with appropriate libraries installe
 ## Rendering PDFs with Typst
 
-MyST also provides an option to build PDFs with [Typst](https://typst.app). Typst is a markup-based typesetting language. Compared to $\LaTeX$, syntax is streamlined and consistent, and compile time is significantly faster. To render Typst PDFs locally, you must install the [Typst CLI](https://github.com/typst/typst).
+[Typst](https://typst.app) is a markup-based typesetting language. Compared to $\LaTeX$, syntax is streamlined and consistent, and compile time is significantly faster.
```
```suggestion
[Typst](https://typst.app) is a markup-based typesetting language. It is **significantly faster and simpler than using $\LaTeX$** with results of equal or better quality.
```

stefanv commented on docs/creating-pdf-documents.md:
```diff
@@ -93,9 +94,31 @@ Ensure that you download a full distribution with appropriate libraries installe
 ## Rendering PDFs with Typst
 
-MyST also provides an option to build PDFs with [Typst](https://typst.app). Typst is a markup-based typesetting language. Compared to $\LaTeX$, syntax is streamlined and consistent, and compile time is significantly faster. To render Typst PDFs locally, you must install the [Typst CLI](https://github.com/typst/typst).
+[Typst](https://typst.app) is a markup-based typesetting language. It is **significantly faster and simpler than using $\LaTeX$** with results of equal or better quality.
```
This removes the context that myst is one of two PDF compilation engines supported by myst; probably still useful context?

choldgraf replied:
good point - I'll add it to the top

stefanv commented on docs/creating-pdf-documents.md:
```diff
@@ -93,9 +94,31 @@ Ensure that you download a full distribution with appropriate libraries installe
+### How to install Typst
+
+Follow [the Typst installation instructions](https://github.com/typst/typst?tab=readme-ov-file#installation) for several options to install Typst.
+We **strongly recommend using the latest releases of Typst**. If you get a confusing Typst error, a good first step is to upgrade your version of Typst.
```
Should we specifically warn against installing with npm, since that is such a common thing to do with javascript packages?

choldgraf commented on docs/creating-pdf-documents.md:
````diff
@@ -108,6 +131,9 @@ exports:
 ---
 ```
 
+You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.
+The Typst templates use the [MyST templating library](xref:jtex) and support the same configuration as [$\LaTeX$](#render-latex).
+
 ## Choosing a Template
````
````suggestion
Finally, build the PDF output with Typst using the following command:

```bash
myst build --typst
```

You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.
The Typst templates use the [MyST templating library](xref:jtex) and support the same configuration as [$\LaTeX$](#render-latex).

## Choosing a Template
````

rowanc1 commented on docs/creating-pdf-documents.md:
```diff
@@ -93,9 +94,31 @@ Ensure that you download a full distribution with appropriate libraries installe
+Follow [the Typst installation instructions](https://github.com/typst/typst?tab=readme-ov-file#installation) for several options to install Typst.
+We **strongly recommend using the latest releases of Typst**. If you get a confusing Typst error, a good first step is to upgrade your version of Typst.
+
+We recommend installing Typst with Mamba, like so:
```
Similar to npm, the conda versions are not maintained by the core team (as far as I can tell). So these will lag behind and/or could be abandoned. My recommendation would be to follow the typst docs directly.

choldgraf replied:
ok will remove that

agoose77 replied:
I'm in favour of keeping mamba instructions. Many packages on conda-forge are community maintained, and it makes thinks a lot easier if users are pulling other things from conda. I think we focus on what works well right now. Unless we find that the version on forge is already too old for our required feature set.

choldgraf replied:
Personally I agree with your point about mamba - I included it originally because it's how we recommend installing myst for Python users, and the conda forge version of typst seemed to be updated.

That said I don't have a lot of mental bandwidth for refining this PR so I'd prefer if we just got this to a stage that was safe to merge and then did it. If folks want to debate whether mamba should make it into the docs then I'm happy for somebody to take over the PR.

choldgraf commented:
I think I addressed all of these comments - I had to update some language throughout to clarify some of these concepts here and there. We will need to take a deeper pass at this page at some point.

stefanv commented on docs/creating-pdf-documents.md:
````diff
@@ -107,6 +129,14 @@ exports:
+myst build --typst
+```
+
+You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.
````
Out of curiosity, is this recommended? `myst build --typst` will probably only work if you have a `myst.yml` file, so isn't that then the place to put the options?

choldgraf replied:
I am not sure - I think a lot of mystmd originated with the assumption that you were working in individual documents / articles, and thus put frontmatter into the `.md` file. The move into the `executablebooks/` org brought another major use-case: multi-page books, and has significantly boosted the importance of `myst.yml`. But I feel like there are still a lot of legacy mentions of article-focused workflows that don't hold anymore.

My intuition is that we should just merge this one for now since I don't really know the answer to your question. I am happy to try and brainstorm an answer to your question w/ you as a follow up.

stefanv reviewed (approved):
Looks like a helpful improvement to me; thanks @choldgraf!

agoose77 commented on docs/creating-pdf-documents.md:
````diff
@@ -107,6 +129,14 @@ exports:
+Finally, build the PDF output with Typst using the following command:
+
+```bash
+myst build --typst
````
For now, I'm going to update this for the single-document workflow despite the points above -- we'll iterate at some point to cover `myst.yml`
```suggestion
myst build article.md --typst
```
